.travis.yml

```diff
@@ -3,6 +3,32 @@ language: cpp
 jobs:
   include:
 
+    - stage: Linter + Doxygen + non-debug Ubuntu/gcc-5 test
+      env: NAME="clang-format"
+      addons:
+        apt:
+          packages:
+            - clang-format-3.8
+      install:
+      script: |
+        # Apparently update-alternatives doesn't work in Travis containers
+        mkdir -p priority-symlinks
+        ln -s /usr/bin/clang-format-3.8 priority-symlinks/clang-format
+        export PATH=${PWD}/priority-symlinks:${PATH}
+
+        # Now we can do the formatting pass
+        clang-format --version
+        git-clang-format-3.8 "${TRAVIS_BRANCH}"
+        git diff --color > formatted.diff
+        if [[ -s formatted.diff ]] ; then
+          echo 'Formatting error! Apply the following diff and resubmit:'
+          cat formatted.diff
+          exit 1
+        fi
+        echo 'No formatting errors found'
+        exit 0
+      before_cache:
+
```

Comment:
I'm unsure as to whether this should be another commit or not.  In some ways it would be nice to roll the tool out locally, let people use it, then add it to CI.

Reply:
I don't really understand the drawbacks of enabling this straight away. Could you give some examples?

Reply:
It may wrongly block other PRs from passing CI.

Reply:
Unless there's a way of enabling travis checks just for new branches, I believe this will be an issue whenever we enable this check. In any case, the formatting can be corrected using the instructions in the COMPILING.md (the 'retroactively formatting individual commits' section).

Reply:
@reuk would this step block a PR? The current linter, if fails, doesn't fail the PR. This shouldn't either, since there may be unforseen problems. 

However, providing it doesn't then I would have thought turning it on in this PR would be the quickest way to find out if there are issues (as opposed to relying on people doing it locally).

Reply:
At the moment it does block, but maybe if we made it non-blocking like the linter that would be a good solution - we can always make it blocking in the future.

Reply:
Done

Reply:
Non-blocking is good.  It seems like the kind of thing that it might take a little while to get used to and to get to love, so making it blocking on day 1 seems a bit hasty.  There may also be unforseen things that we need to iron out and I'd rather not block commits on CI while that happens.